extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -846,4 +852,48 @@ export class AmpA4A extends AMP.BaseElement {
       this.stylesheets_.push(linkElem);
     });
   }
+
+  /**
+   * Handle click on links and replace variables in the click URL.
+   * @param {!Event} e
+   */
+  handleClick_(e) {
+    const target = closestByTag(dev().assertElement(e.target), 'A');
+    if (!target || !target.href) {
+      // Not a click on a link.
+      return;
+    }
+    const hrefToExpand =
+    target.getAttribute(ORIGINAL_HREF_ATTRIBUTE) || target.getAttribute('href');
+    if (!hrefToExpand) {
+      return;
+    }
+    const vars = {
+      // The click location assume a switch from ShadowDOM to friendly iframe
+      // based rendering, in which case the document relative positions are
+      // really ad-relative.
+      'CLICK_X': () => {
+        return e.pageX;
```

Comment:
Given that we're not handling ad offset correctly here in the shadow DOM case, I'm concerned that we may inadvertently trip spam filters due to invalid click x/y values being sent.  I propose we either wait for PR #5196 before submitting (will need to handle conflicts) or update the X/Y to use the shadow DOM offset as I had done previously and then I will remove as part of PR #5196